pkg/asset/machines/openstack/openstackmachines.go

```diff
@@ -179,30 +181,31 @@ func generateMachineSpec(clusterID string, platform *openstack.Platform, mpool *
 		securityGroups = append(securityGroups, capo.SecurityGroupFilter{ID: securityGroup})
 	}
 
-	// FIXME: Uncomment when the server group rework merged
-	// https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1779
-	// serverGroupName := clusterID + "-" + role
 	spec := capo.OpenStackMachineSpec{
-		CloudName: CloudName,
-		Flavor:    mpool.FlavorName,
+		Flavor: mpool.FlavorName,
 		IdentityRef: &capo.OpenStackIdentityReference{
-			Kind: "Secret",
-			Name: clusterID + "-cloud-config",
+			Name:      clusterID + "-cloud-config",
+			CloudName: CloudName,
+		},
+		Image: capo.ImageFilter{
+			Name: osImage,
 		},
-		// FIXME(stephenfin): We probably want a FIP for bootstrap?
-		// TODO: This is an image name. Migrate to a filter with Name when API v1alpha8 is released.
-		Image:          osImage,
 		Ports:          append([]capo.PortOpts{port}, additionalPorts...),
 		SecurityGroups: securityGroups,
-		// FIXME: Uncomment when the server group rework merged
-		// https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1779
-		//ServerGroup: *capo.ServerGroupFilter{
-		//	"Name": serverGroupName,
-		// },
-		ServerMetadata: map[string]string{
-			"Name":               fmt.Sprintf("%s-%s", clusterID, role),
-			"openshiftClusterID": clusterID,
+		ServerGroup: &capo.ServerGroupFilter{
```

Comment:
I stared seeing the following issue:
`E0313 17:05:59.126881 1270406 controller.go:329] "Reconciler error" err="no server group with name test-s7ppd-master could be found" controller="openstackmachine" controllerGroup="infrastructure.cluster.x-k8s.io" controllerKind="OpenStackMachine" OpenStackMachine="openshift-cluster-api-guests/test-s7ppd-master-2" namespace="openshift-cluster-api-guests" name="test-s7ppd-master-2" reconcileID="f85ab843-4fdf-4a52-887f-5f1faba8e772"`

The sever group [support existent on CAPO only filters a sever group](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/docs/book/src/topics/crd-changes/v1alpha7-to-v1beta1.md#change-to-servergroupid), so we must pre-create this in the installer, let me know in case you plan to handle it or I could help.

Reply:
I'm on it, thanks!

Reply:
https://github.com/openshift/installer/pull/8164